src/vstest.console/Processors/TestAdapterLoadingStrategyArgumentProcessor.cs

```diff
@@ -0,0 +1,276 @@
+            }
+
+            return _metadata;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the executor.
+    /// </summary>
+    public Lazy<IArgumentExecutor> Executor
+    {
+        get
+        {
+            if (_executor == null)
+            {
+                _executor = new Lazy<IArgumentExecutor>(() => new TestAdapterLoadingStrategyArgumentExecutor(CommandLineOptions.Instance, RunSettingsManager.Instance, ConsoleOutput.Instance, new FileHelper()));
+            }
+
+            return _executor;
+        }
+
+        set
+        {
+            _executor = value;
+        }
+    }
+}
+
+/// <summary>
+/// The argument capabilities.
+/// </summary>
+internal class TestAdapterLoadingStrategyArgumentProcessorCapabilities : BaseArgumentProcessorCapabilities
+{
+    public override string CommandName => TestAdapterLoadingStrategyArgumentProcessor.CommandName;
+
+    public override bool AllowMultiple => false;
+
+    public override bool IsAction => false;
+
+    public override bool AlwaysExecute => true;
+
+    public override ArgumentProcessorPriority Priority => ArgumentProcessorPriority.TestAdapterLoadingStrategy;
+
+    public override string HelpContentResourceName => CommandLineResources.TestAdapterLoadingStrategyHelp;
+
+    public override HelpContentPriority HelpPriority => HelpContentPriority.TestAdapterLoadingStrategyArgumentProcessorHelpPriority;
+}
+
+/// <summary>
+/// The argument executor.
+/// </summary>
+internal class TestAdapterLoadingStrategyArgumentExecutor : IArgumentExecutor
+{
+    #region Fields
+    /// <summary>
+    /// Used for getting sources.
+    /// </summary>
+    private readonly CommandLineOptions _commandLineOptions;
+
+    /// <summary>
+    /// Run settings provider.
+    /// </summary>
+    private readonly IRunSettingsProvider _runSettingsManager;
+
+    /// <summary>
+    /// Used for sending output.
+    /// </summary>
+    private readonly IOutput _output;
+
+    /// <summary>
+    /// For file related operation
+    /// </summary>
+    private readonly IFileHelper _fileHelper;
+
+
+    private static readonly string[] EmptyStringArray =
+#if NET451
+                                                        new string[0];
+#else
+                                                        Array.Empty<string>();
+#endif
+
+    #endregion
+
+    public const string RunSettingsPath = "RunConfiguration.TestAdapterLoadingStrategy";
+    public const string DefaultStrategy = "Default";
+    public const string ExplicitStrategy = "Explicit";
+
+    /// <summary>
+    /// Default constructor.
+    /// </summary>
+    /// <param name="options"> The options. </param>
+    /// <param name="testPlatform">The test platform</param>
+    public TestAdapterLoadingStrategyArgumentExecutor(CommandLineOptions options, IRunSettingsProvider runSettingsManager, IOutput output, IFileHelper fileHelper)
+    {
+        Contract.Requires(options != null);
```

Comment:
are we using code contracts yet?

Reply:
I am not sure. I just copied another argument executor :)

Reply:
remove it